test/runtime/src/lib.rs

```diff
@@ -295,7 +297,14 @@ decl_runtime_apis! {
 impl_runtime_apis! {
 	impl sp_api::Core<Block> for Runtime {
 		fn version() -> RuntimeVersion {
-			VERSION
+			if VersionUpgrade::get() {
```

Comment:
This will not work. You need to use some value that is set at genesis and removed at the runtime upgrade.

And here you need to check if this value isn't set.

Reply:
Ok I got it. Like this 1215f65 ?